src/pages/docs/[[...slug]].tsx

```diff
@@ -29,31 +28,45 @@ import {
 import { Label } from '../../components/common/Label';
 import { DocsFooter } from '../../components/docs/DocsFooter';
 import { Callout } from '../../components/common/Callout';
+import { TreeNode } from 'types/TreeNode';
 
-type Ctx = GetStaticPropsContext<{
+const redirects = [
+  { from: 'get-started/app-lifecycle', to: 'core-concepts/app-lifecycle' },
+  { from: 'get-started/offline-support', to: 'core-concepts/offline-support' },
+  { from: 'inputs/button-group', to: 'core-concepts/inputs/button-group' },
+  { from: 'inputs/multi-select', to: 'core-concepts/inputs/multi-select' },
+  { from: 'inputs/number', to: 'core-concepts/inputs/number' },
+  { from: 'inputs/select', to: 'core-concepts/inputs/select' },
+  { from: 'inputs/text', to: 'core-concepts/inputs/text' },
+  { from: 'inputs/toggle', to: 'core-concepts/inputs/toggle' },
+];
+
+type Ctx = GetServerSidePropsContext<{
   slug?: string[];
 }>;
 
-export async function getStaticPaths() {
-  const paths = allDocs
-    .map((d) => d.pathSegments.map((pS: PathSegment) => pS.pathName).join('/'))
-    .map(toParams);
-
-  return {
-    paths,
-    fallback: false,
-  };
-}
-
-export const getStaticProps = async (ctx: Ctx) => {
+export const getServerSideProps = async (ctx: Ctx) => {
   const { params } = ctx;
   const pagePath = params?.slug?.join('/') ?? '';
 
   const doc = allDocs.find(
     (d) =>
       d.pathSegments.map((pS: PathSegment) => pS.pathName).join('/') ===
       pagePath,
-  )!;
+  );
+
+  // If doc is not found, check if pagePath is a redirect.
+  if (!doc) {
+    const redirect = redirects.find((r) => r.from === pagePath);
+    if (redirect) {
+      return {
+        redirect: { destination: `/docs/${redirect.to}`, permanent: true },
+      };
+    } else {
+      throw new Error('Doc not found: ' + pagePath);
+    }
+  }
+
```

Comment:
Switching from Next.JS' [static generation](https://nextjs.org/docs/basic-features/data-fetching/get-static-props) to [server-side ](https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props) to support redirects.